srcs/app/http/HttpRequestHandler.cpp

```diff
@@ -271,16 +274,64 @@ void	HttpRequestHandler::DoGet_(const HttpRequest &request) {
 	}
 }
 
-void	HttpRequestHandler::DoPost_(const HttpRequest &request) {
-	// TODO(any) Implement POST
-	(void)request;
-	HttpResponse	response(200);
-	std::string		body = "Responding to a POST request\n";
+bool	HttpRequestHandler::IsCGI_(const std::string &full_path) const {
+	const std::string extension = PathExtension_(full_path);
+	return request_location_->common.cgi_assign.count(extension) > 0;
+}
 
-	response.AddHeader("Content-Type", "text/plain");
-	response.SetBody(body);
-	AddCommonHeaders_(&response);
-	raw_response_ = response.CreateResponseString();
+bool	HttpRequestHandler::IsUploadEnabled_() const {
+	return !request_location_->common.upload_store.empty();
+}
+
+bool	HttpRequestHandler::IsValidUploadPath_(const std::string &path) const {
+	if (request_location_->HasLocation()) {
+		return path == *request_location_->path;
+	}
+	return path == "/";
+}
+
+void	HttpRequestHandler::UploadFile_(const HttpRequest &request) {
+	try {
+		FormFile form_file(request);
+		const std::string full_upload_path =
+								request_location_->common.upload_store +
+								form_file.GetFilename();
+		std::ofstream out(full_upload_path.c_str());
+		if (!out) {
+			PathError_();
+			return;
+		}
+		const std::string file_content = form_file.GetFileContent();
+		out.write(file_content.c_str(), file_content.size());
+		if (!out) {
+			RequestError_(500);
+		} else {
+			DefaultStatusResponse_(200);
+		}
+	}
+	catch (const std::exception &e) {
+		RequestError_(400);
+	}
+}
+
+void	HttpRequestHandler::DoPost_(const HttpRequest &request) {
+	const std::string request_path = request.GetPath();
+	const std::string full_path =
+							request_location_->common.root + request_path;
+	if (IsRegularFile_(full_path)) {
+		if (!IsCGI_(full_path)) {
+			RequestError_(404);
+		} else {
+			// TODO(any) Implement CGI
+			RequestError_(501);
+		}
+	} else {
+		if (!IsUploadEnabled_() || !IsValidUploadPath_(request_path)) {
+			RequestError_(403);
+		} else {
+			UploadFile_(request);
+		}
```

Comment:
Think possitively, be optimistic! :smile: 

```suggestion
		if (IsUploadEnabled_() && IsValidUploadPath_(request_path)) {
			UploadFile_(request);
		} else {
			RequestError_(403);
		}
```